lib/metasploit/credential/importer/core.rb

```diff
@@ -140,57 +134,105 @@ def import_long_form
-                                                     realm: realm_object_for_row )
-
-        # Make Logins with attendant Host/Service information if we are doing that
-        if host_address.present? && service_port.present? && service_protocol.present?
-          login_opts = {
-            core: core,
-            address: host_address,
-            port: service_port,
-            protocol: service_protocol,
-            workspace_id: workspace.id,
-            service_name: service_name.present? ? service_name : ""
-          }
-          login_opts[:last_attempted_at] = last_attempted_at unless status.blank?
-          login_opts[:status]            = status unless status.blank?
-          login_opts[:access_level]      = access_level unless access_level.blank?
-
-          create_credential_login(login_opts)
+            create_credential_login(login_opts)
+
+          end
         end
       end
-    end
+      end
+    return all_creds_valid
   end
 
 
   # Performs an import of a "short" form of CSV - one that contains only one type of {Metasploit::Credential::Private}
   # and no {Metasploit::Credential::Realm} data
-  # @return [void]
+  # @return [Boolean]
   def import_short_form
+    core_opts = []
+    all_creds_valid = true
     Metasploit::Credential::Core.transaction do
       csv_object.each do |row|
         next if row.header_row?
 
-        username     = row['username']
-        private_data = row['private_data']
+        username     = row['username'].present? ? row['username'] : ''
+        private_data  = row['private_data'].present? ? row['private_data'] : ''
 
         public_object = create_public_from_field(username)
 
         if private_data.strip == BLANK_TOKEN
           private_object_for_row = Metasploit::Credential::BlankPassword.first_or_create
         else
-          private_object_for_row = private_credential_type.constantize.where(data: row['private_data']).first_or_create
+          private_object_for_row = @private_credential_type.constantize.where(data: row['private_data']).first_or_create
         end
 
-        create_credential_core(origin:origin, workspace_id: workspace.id,
+        # need to check private_object_for_row.valid? to raise a user facing message if any cred had invalid private
+
+        all_creds_valid = all_creds_valid && (public_object.valid? && private_object_for_row.valid?)
+
+
+        core_opts << {origin:origin, workspace_id: workspace.id,
                                       public: public_object,
-                                      private: private_object_for_row)
+                                      private: private_object_for_row}
       end
+      if all_creds_valid
+        core_opts.each do |item|
+          if Metasploit::Credential::Core.where(private:item[:private]).blank?
+            create_credential_core(item)
+          end
+        end
+      end
+
+
     end
+    return  all_creds_valid
```

Comment:
No need to return this. We can call valid? on the instance of this object and have all the validators run. It would achieve the same thing.